examples/aggregate_signature/helpers.rs

```diff
@@ -0,0 +1,102 @@
+use crate::aggregate_signature::registration::Registration;
+use crate::aggregate_signature::signature::IndividualSignature;
+use crate::{AlbaThresholdSignature, Element};
+use blake2::digest::{Update, VariableOutput};
+use blake2::Blake2bVar;
+use blst::min_sig::{AggregatePublicKey, AggregateSignature, PublicKey, Signature};
+use blst::BLST_ERROR;
+use std::collections::HashMap;
+
+/// Helper function to compute a commitment by hashing `(checksum || msg)`
+pub(crate) fn get_commitment<const N: usize>(checksum: &[u8], msg: &[u8]) -> [u8; N] {
+    let mut hasher = Blake2bVar::new(N).expect("Invalid hash size");
+    let mut commitment = [0u8; N];
+
+    hasher.update(checksum);
+    hasher.update(msg);
+    hasher
+        .finalize_variable(&mut commitment)
+        .expect("Hash finalization failed");
+    commitment
+}
+
+/// Collect valid signatures by verifying each individual signature.
+/// Returns a hashmap of valid signature bytes to their corresponding index.
+pub(crate) fn collect_valid_signatures<const N: usize>(
+    signature_list: &[IndividualSignature],
+    registration: &Registration,
+    msg: &[u8],
+) -> HashMap<Element, usize> {
+    let mut valid_signatures = HashMap::new();
+
+    for sig in signature_list {
+        if let Some(verification_key) = registration.registered_keys.get(&sig.index) {
+            let Some(checksum) = &registration.checksum else {
+                println!("Error: Registration is not closed. Cannot verify signatures.");
```

Comment:
You might print this lines many times, perhaps you could move it outside the loop?

Reply:
We might also not collect the signatures if the registration is not closed (Unless you want to show something in a future example)?